tests/sqlparser_clickhouse.rs

```diff
@@ -626,6 +626,24 @@ fn parse_create_materialized_view() {
     clickhouse_and_generic().verified_stmt(sql);
 }
 
+#[test]
+fn parse_group_by_with_modifier() {
+    match clickhouse_and_generic().verified_stmt("SELECT * FROM t GROUP BY x WITH ROLLUP WITH CUBE")
```

Comment:
For the tests, following cases I think we can cover

- We enumerate all the WITH modifiers we introduced so that they covered by some test
- We add `GROUP BY ALL WITH ...` examples
For above would probably make sense to use a table of scenarios to enumerate e.g pseudocode

```
let clauses = ["x", "a, b", "ALL"]
let modifiers = ["WITH ROLLUP", "WITH CUBE", "WITH ROLLUP WITH CUBE" ... ]
for clause in clauses
    for modifier in modifiers
         clickhouse_and_generic().verified_stmt!("SELECT * FROM t GROUP BY [clause} {modifier})
         ...
```

- Negative tests e.g `GROUP BY x WITH ROLLUP WITH`, `GROUP BY x WITH WITH ROLLUP`, `GROUP BY WITH ROLLUP`